gef.py

```diff
@@ -8658,81 +8658,96 @@ def get_settings_path(self):
         return path if os.path.isdir(path) else None
 
 
-class GenericOffsetFunction(gdb.Function):
+@lru_cache()
+def get_section_base_address(name):
+    section = process_lookup_path(name)
+    if section:
+        return section.page_start
+
+    return None
+
+@lru_cache()
+def get_zone_base_address(name):
+    zone = file_lookup_name_path(name, get_filepath())
+    if zone:
+        return zone.zone_start
+
+    return None
+
+class GenericFunction(gdb.Function):
     """This is an abstract class for invoking offset functions, should not be instantiated."""
     __metaclass__ = abc.ABCMeta
 
-    @abc.abstractproperty
-    def _section_(self): pass
-    @abc.abstractproperty
-    def _zone_(self): pass
     @abc.abstractproperty
     def _function_(self): pass
     @property
     def _syntax_(self):
         return "${}([offset])".format(self._function_)
 
     def __init__ (self):
-        super(GenericOffsetFunction, self).__init__(self._function_)
+        super(GenericFunction, self).__init__(self._function_)
 
     def invoke(self, offset=gdb.Value(0)):
         if not is_alive():
             raise gdb.GdbError("No debugging session active")
 
         base_address = self.get_base_address()
-        if not base_address:
-            raise gdb.GdbError("No {} section".format(self._section_ or self._zone_))
-
         addr = long(offset) if offset.address is None else long(offset.address)
+
         return long(base_address + addr)
 
-    def get_base_address(self):
-        if self._section_:
-            section = process_lookup_path(self._section_)
-            if section:
-                return section.page_start
-        elif self._zone_:
-            zone = file_lookup_name_path(self._zone_, get_filepath())
-            if zone:
-                return zone.zone_start
-        return None
+    @abc.abstractmethod
+    def get_base_address(self): pass
```

Comment:
I don't think this should be the name of the function. In fact, I don't think a generic function should define a required function name, unless it's `do_invoke` like with `GenericCommand`